<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>CompBio ToDo Notes</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" />
</head>
<body>

    
    <div class="navbar navbar-static-top">
    <div class="navbar-inner">
      <div class="container">
        <span class="doc-title">CompBio ToDo Notes</span>
        <ul class="nav pull-right doc-info">
                            </ul>
      </div>
    </div>
  </div>
    <div class="container">
    <div class="row">
            <div id="TOC" class="span3">
        <div class="well toc">

        <ul>
        <li><a href="#to-do-weekly" id="toc-to-do-weekly">To Do
        (weekly)</a></li>
        <li><a href="#take-into-account-for-future"
        id="toc-take-into-account-for-future">Take into account for
        future</a></li>
        <li><a href="#done" id="toc-done">Done</a></li>
        </ul>

        </div>
      </div>
            <div class="span9">

      
      <h1 id="to-do-weekly">To Do (weekly)</h1>
<ul>
<li><label><input type="checkbox" /><strong>Improve annotation for
genes:</strong> I missed the use of tools like InterProScan.</label>
<ul>
<li>During a meeting Eric Ross and Sukha Malladi annotate using their
pipeline intended for eukaryotes, but has some tools useful for bacteria
like interproscan, nonetheless they did not find significant matches for
hypothetical proteins.</li>
</ul></li>
<li>[x]
<ul>
<li>On the Illumina MAG strains.
<ul>
<li>I will continue documenting the strategy here <a href="Plasmidome"
class="wikilink">Plasmidome</a></li>
<li>I used mob-typer to know the genomes that have 3 or more relaxases
and I mapped to the chromosomes, and super-plasmids to get the unmapped
sequences as putative accessory plasmids sequences.</li>
<li><label><input type="checkbox" /><strong>There are some sequences
with 0 relaxases predictions. What strategy can we use there? Possible
strategies to explore:</strong></label>
<ul>
<li><label><input type="checkbox" />I will try to see percentage of
similarity of RepABC, it is apparently conserved in the accessory
plasmids, how different it is from the super-plasmids???</label>
<ul>
<li><strong>My preliminary observations</strong> say that RepABC differ
significantly between accessory plasmids and mega-plamids. However, for
example in mega plasmid I have found RepA duplication with significant
identity divergence. <strong>Therefore, I can only count on finding
RepABC in the same contig.</strong></li>
</ul></li>
<li>I saw that it can happen that relaxases were zero, however it still
found the MPF, but I cannot use number of MPF machinery as some
accessory plasmids can have it, so I cannot difference between psymA/B
and accessory. Additionally, this comprises repetitions that very likely
will be hard for most Illumina assemblies.</li>
<li>I could try to keep cleaning the “unique” alignments. For example,
GCA_033712535 presents zero relaxases, and it presents a 64kb putative
plasmid sequence, when checking at mapping it is sharing 44% with a
psymA reference. <strong>It’s probably that it belongs to it given the
variability described for this super-plasmid</strong>, so I could try
annotate Genomic Islands (GIs), apparently there are tool to predict
genes acquired horizontally, so if I can determine this for this chunk I
could rule it out. Does this have to do with insertion sites
(ISs)???</li>
</ul></li>
</ul></li>
</ul></li>
<li><label><input type="checkbox" />Build a Pan-genome graph for S.
meliloti using susceptible strains. I have to wait for Madi
info.</label>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />I received the
susceptible MAG from Madi on Jul. 14. <a
href="https://webfs/n/projects/jp2992/MOLNG4331/resistant_strains.xlsx">resistant_strains.xlsx</a></label></li>
<li><label><input type="checkbox" checked="" />PPanGGoLin for pan-genome
construction. <em>Update:</em> Its faster than Roary, it has support and
plenty of several stats and graphs.</label></li>
<li><label><input type="checkbox" />Try minigraph-cactus as an
alternative to progressiveMauve since I do not like the output of it. it
also allows to create pan-genome graph.</label></li>
</ul></li>
<li><label><input type="checkbox" /><strong>Regarding the protein vs
protein comparison between KH35cA and pMAG215.</strong> They are very
divergent, a mmseqs2 comparison or BLASTP is no enough to get all the
homologous. Here an example:</label>
<ul>
<li>Both plasmids were annotated for <em>S. meliloti</em> RepABC,
however only RepA/C are clustered under thresholds for 80-90% cov and
50-80% id. RepB is very divergent at the point that they are predicted
under a different non-redundant sequence assigned to <em>S.
meliloti</em>. For example, for KH35cA is WP_017272063.1 and for pMAG215
is WP_014989225.1. I think MMseqs2 offers a possible approach which
would involve using an external database this strategy might work as it
implies using other sequences that might work better as bait to cluster
both of them. But I think it would be too much for only 2 sequences.
<strong>Maybe a functional annotation comparison between both would be
better?</strong> <img src="Screenshot 2025-06-30 at 1.40.15 PM.png"
class="wikilink" alt="Screenshot 2025-06-30 at 1.40.15 PM.png" /></li>
</ul></li>
<li><label><input type="checkbox" checked="" /><strong>Assemble the 121
strains from collaborator.</strong> <em>Update:</em> There are only 8
files. Something wrong happened. Siva will take care of it.</label></li>
</ul>
<p>Future: - [ ] Pan-genome of everything. - [ ] Annotate SNPs using
DeepVariant or Clair3. - [ ] Annotate effects with SnpEff.</p>
<h1 id="take-into-account-for-future">Take into account for future</h1>
<ul>
<li><figure>
<img src="Plasmidome" class="wikilink" alt="Plasmidome" />
<figcaption aria-hidden="true">Plasmidome</figcaption>
</figure></li>
<li></li>
</ul>
<h1 id="done">Done</h1>
<ul>
<li><p><label><input type="checkbox" checked="" /><strong>Classify
strains in collaborator with acce. plasmids to run on
gel</strong>.</label></p>
<ul>
<li>Among the best plasmids (check report), I will classify the
assembled plasmids as super good, intermedia and fishy.</li>
<li>I have to do the same for the illumina MAG strains.</li>
</ul></li>
<li><p><label><input type="checkbox" checked="" /><strong>Recover
plasmids.</strong></label></p>
<ul>
<li>I downloaded the last 2024 version of PLSDB database, which is a
curated database of plasmid (only complete sequences) here our number of
plasmids for meliloti is reduced to 57.<br />
</li>
<li>On collab. assemblies.
<ul>
<li>When checking at the biomarkers by mob_suite for all the plasmids in
the plsdb, I found the following frequencies for relaxases:
<ul>
<li><strong>Plasmids greater than 500kb:</strong>
<ul>
<li>MOBP: 3</li>
<li>MOBQ: 99</li>
</ul></li>
<li><strong>Plasmids lower than 500kb:</strong>
<ul>
<li>MOBV: 1</li>
<li>MOBQ: 42</li>
<li>MOBP: 19</li>
</ul></li>
</ul></li>
<li>This analysis came up after me thinking that for psymA/B it was
always predicted MOPQ and for accessory plasmids I could find MOBP/Q
however for example in 2002 if I found a mega-plasmid with no relaxase
prediction and one with both MOBPQ/P.</li>
<li>What I want to know is if this can work as a metric to find
miss-assembled PsymA/B and putative acce. plasmids.</li>
</ul></li>
</ul></li>
<li><p><label><input type="checkbox" checked="" /><strong>Define a
coverage.</strong> To do this, I will:</label></p>
<ul>
<li>Define which assembly works better, the collaborator data contains
coverage from approx. 10x to 50x very uneven between contigs and with
reads sizes two times smaller than ours.
<strong><em>Update:</em></strong> The low coverage cause in flye two
interesting problems: 1, fragment genomes by repetition, and 2, mix
accessory plasmid and super plasmid by repetition. Hifiasm was bolder,
it miss-assembled mega-plasmid and accessory plasmids. Check section of
collaborator assembly.</li>
</ul></li>
<li><p><label><input type="checkbox" checked="" /><strong>For Madi
presentation</strong>:</label></p>
<ul class="task-list">
<li><p><label><input type="checkbox" checked="" />Dot plot in house
plasmid vs plasmid:
https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/homology/all_vs_pmag215/out.pdf</label></p></li>
<li><p><label><input type="checkbox" checked="" />Matrix accessory genes
for plasmids (80% identity):
https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/roary/roary_result_80_pct/pangenome_matrix.png</label></p></li>
<li><p><label><input type="checkbox" checked="" />Matrix accessory genes
for whole genome (80% identity):
https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/roary_result_80_pct_1751912805/pangenome_matrix.png</label></p></li>
<li><p><label><input type="checkbox" checked="" />Pie accessory genes
for whole genome (80% identity):
https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/roary_result_80_pct_1751912805/pangenome_pie.png</label></p></li>
<li><p><label><input type="checkbox" checked="" />Dot plot for whole
genomes:
https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/minimap2/all_vs_mag215_asm20.paf.pdf</label></p></li>
<li><p><label><input type="checkbox" checked="" />Synteny plot between
in-house strains:
https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/minimap2/plot_synteny/ngenomesyn/prueba.svg</label></p></li>
<li><p><label><input type="checkbox" checked="" />Can we find out which
of these illumina strains have accessory plasmids and then do PacBio
assembles of them?</label></p></li>
<li><p><label><input type="checkbox" checked="" />Look into average
coverage per assembly, compare to recommended coverage for bacteria, how
many samples can we sequence at once and have enough Coverage for
assembly?<br />
</label></p></li>
<li><p><label><input type="checkbox" checked="" />Run de-novo ORI
prediction on all our new plasmids.</label></p></li>
</ul></li>
<li><p>30/06/2025 - 09/07/2025</p>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />Nuclear genome: Process
the 8 PacBio resistant strains</label>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />Check Quality</label>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />Contiguity</label></li>
<li><label><input type="checkbox" checked="" />Completeness with
CheckM.</label></li>
<li><label><input type="checkbox" checked="" />Descrip. stats with
SeqKit.</label></li>
<li><label><input type="checkbox" checked="" />With Flye report I
realized A02 assembly has a shared site between the main chromosome and
the putative plasmid. The assembler decided to separate them. <strong>I
will assemble with Hifiasm</strong>. <em>Update:</em> Hifiasm was abled
to solved the ambiguity.</label></li>
</ul></li>
<li><label><input type="checkbox" checked="" />Genome to genome
comparison.</label></li>
<li><label><input type="checkbox" checked="" />Gene presence/absence
between all genomes (pan-genome). Get unique genes to
MAG215.</label></li>
</ul></li>
<li><label><input type="checkbox" checked="" />Accessory plasmids:
Process the new plasmids if apply.</label>
<ul>
<li><label><input type="checkbox" checked="" />Typing plasmids with mob
suite</label>
<ul>
<li>There is a plasmid non-mobilizable, which casually is the one with
weird assembly graph (A02). <strong>meaning that maybe it is not a
plasmid but a repetition in the chromosome</strong>.</li>
</ul></li>
<li><label><input type="checkbox" checked="" />Plasmid to plasmid
comparison.</label></li>
<li><label><input type="checkbox" checked="" />Check homology between
these plasmids and public plasmids at ncl level.</label></li>
<li><label><input type="checkbox" checked="" />Check homology between
these plasmids and MAG215 at ncl level.</label></li>
<li><label><input type="checkbox" checked="" />Roary to get shared genes
between in-house plasmids. Get unique genes to pMAg215.</label></li>
<li><label><input type="checkbox" checked="" />I could try check Promer
(from MUMmer) to do all-vs-all genome comparison at protein level to
check synteny blocks. <em>Update:</em> there was no much difference with
nucleotide alignment.</label></li>
<li>[ ]</li>
</ul></li>
</ul></li>
<li><p>27/06/2025</p>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />  <strong>Compare
accessory plasmid’s genes of KH35c and MAG215</strong>.</label>
<ul>
<li>Siva <em>realized one of our favorite strain KH35c has an accessory
plasmid ~180 kb. Some of the genes I was interested in MAG215 are
present in this accessary plasmid. Experimentally we see some similar
plant phenotype between KH35c and MAG215.</em></li>
<li><label><input type="checkbox" checked="" />Compare at genome level.
<em>Update:</em> I used Minimap2 (asm5) and BLAST.</label></li>
<li><label><input type="checkbox" checked="" />Compare at protein level.
<em>Update:</em> I will use Roary with 80% and 50% of identity and back
up with mmseqs2. This because I like the output format of roary but I
have more intervention in mmseqs. I set up 90% cov 80% id.</label></li>
</ul></li>
</ul></li>
<li><p>26/06/2025</p>
<ul>
<li>I sent Siva report with the alignments of accessory plasmids.</li>
<li>I organized reports</li>
<li>I get vaccinated for hep. B</li>
<li><label><input type="checkbox" checked="" /><strong>Assembly the 8
PacBio resistant strains</strong></label></li>
<li><label><input type="checkbox" checked="" />Check
Quality</label></li>
<li><label><input type="checkbox" />Check accessory plasmid in the 8
resistant strains.</label></li>
</ul></li>
<li><p>25/06/2025</p>
<ul class="task-list">
<li><label><input type="checkbox" checked="" /><strong>Compare MAG215
with MAG assemblies:</strong> After evaluation with MMseqs2. We decided
to try with more closely related strains, apparently MAG215 has diverged
significantly. Siva notified that there were some MAG assemblies
available from other authors and from which they had information about
which ones where susceptible to the peptide.</label>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />Adding Rm1021 and Mag215
to the jaccard comparison. <em>Update:</em> Rm1021 once again shows not
to be the better genome to compare.</label></li>
<li><label><input type="checkbox" checked="" />Compare genome to genome
between mag215 and the closest strain. <em>Update:</em> QUAST comparison
available in the pan-genome report. There is slightly higher coverage in
other MAG strains than with Rm1021.<br />
</label></li>
<li><label><input type="checkbox" checked="" />Compare the 89 strains
and give Madi the closest strain to see if they are susceptible.
<em>Update:</em> Table annotated in pan-genome report.</label></li>
<li><label><input type="checkbox" checked="" />Check if any of the 89
strains have the accessory plasmid from mag215. <em>Update:</em> Table
available at pan-genome report. The strain with the highest proportion
of the plasmid is MAG97. The alignment can be seen in the same QUAST
report as before.</label></li>
</ul></li>
</ul></li>
<li><p>24/06/2025</p>
<ul>
<li>After the jaccard similarity with sourmash and discussion with Siva
and Madi. It looks like the susceptible MAG strains that Madi gave me
are still very divergent. I will make a comparison with all 89 MAG
strains and will include H01 and B02 to see which of them is the
closest.</li>
<li><label><input type="checkbox" checked="" /><strong>Run PGAP on our
two assembled strains</strong>. <em>Update:</em> reduction of genes in
B02 (1021) congruent with NCBI annotation.</label></li>
</ul></li>
<li><p>23/06/2025</p>
<ul>
<li>I keep working on MAG strains.</li>
<li>Madi gave me only 4.</li>
<li>I was abled to get MAG strains from Daniel</li>
</ul></li>
<li><p>20/06/2025</p>
<ul class="task-list">
<li><label><input type="checkbox" checked="" />Drop required similarity
required to 90% and 80% in Roary - how does that change our rm1021 vs
mag215 comparison.</label>
<ul>
<li><strong>It did not change the result significantly</strong>. I tried
50% and it returned a reduction of 254 genes.</li>
</ul></li>
<li><label><input type="checkbox" checked="" />I will inspect the
proteins clustering using MMseqs2. As alternate method, I will cluster
the both proteomes initially using cov 90% id. 90% as NCBI.</label>
<ul>
<li>I tried 90% coverage and 90%, 80% and 50% identity. <strong>There
was a small reduction</strong>. From 90% to 50% there was a reduction of
accessory genes of 220.</li>
</ul></li>
</ul></li>
<li><p>18/06/2025</p>
<ul>
<li>[-] I want to try not splitting the paralogs.
<ul>
<li>There was not a big difference in number of group (4 less). As
expected the greatest change was with the transposes grouping.</li>
</ul></li>
<li>[-] Make snippy call first removing the plasmid. This because the
plasmid is not part itself of the core genome as the super plasmids
always found in smeliloti.</li>
</ul></li>
<li><p>17/06/2025</p>
<ul>
<li>I presented result to Andrew, he approved them.</li>
<li>Some things to finish before meeting with Siva:
<ul>
<li>[-] Proteome comparison between MAG215 and RM1021 using pan-genome
strategy with Roary.
<ul>
<li>[-] Separate files for genes only present in each strain.</li>
<li>[-] Table with genes in the plasmid</li>
</ul></li>
<li>[-] Annotate SNPs in core genome.
<ul>
<li>I used snippy and the core alignment done by Roary and filtered SNPs
with snp-sites.</li>
<li>I want to annotate the output of snp-sites with genes.</li>
</ul></li>
</ul></li>
</ul></li>
<li><p>16/06/2025</p>
<ul>
<li>I set up Roary and ran it using RM1021 and MAg215 to do pan-genome
analysis. The idea is to check absence/presence of genes between both
and I think I can core alignment in the same tool and characterize
variants.</li>
<li>I am trying to functionally annotate the extra plasmid using KEGG
and GO databases. I am a little tangled with this task. BAKTA supposed
to offer grouping by GOG but it did not work I do not the reason. I will
try to do this with other tool.</li>
</ul></li>
<li><p>13/06/2025</p>
<ul>
<li>I downloaded 753 sequences related with the keywords “Sinorhizobium
meliloti[All Fields] AND plasmid[filter]”, after I filtered the
sequences for all greater than 10kb and lower than 500kb to remove the
two big plasmids, the chromosome and genes or cassettes. at the end
there were 114 sequences. I found no similar among 75 sequences mapped,
only short regions of probably homology. This looks like a new plasmid.
The sequences compiled contained also <strong>pSmeSM11a,
pRmeGR4b</strong>.</li>
</ul></li>
<li><p>12/06/2025</p>
<ul>
<li>There are some well described plasmid such as <strong>pSmeSM11a,
pRmeGR4b and pTA2</strong> (I did not find the last one in the
NCBI).</li>
<li>Siva and Medi got 121 new strains we will assemble.</li>
</ul></li>
<li><p>11/06/2025</p>
<ul>
<li>Meeting with Dr. Siva finished with some tasks. I will write them in
the to do list.</li>
<li>QUAST comparison shows that the reference assembly was correctly
assembled and the MAG215 might actually carry a high level of
divergence. A synteny analysis with ProgressiveMauve indicates that
MAG215 (H01) genome might carry acquired and lost particular section
from the reference genome, additionally there are not indications of the
extra plasmid in the reference genome.</li>
</ul></li>
<li><p>10/06/2025</p>
<ul>
<li>I shared advances with my mentor Andrew. While I focused on extra
plasmid annotation, he went for assembly vs assembly comparison between
the NCBI reference and our strain MAG215 and RM1021 (same strain as
NCBI). Also, he did annotation of the genome using LiftOver from NCBI
reference to RM1021 as well as to MAG215, however the latter showed
around 600 genes were not mapped. He will to do the process backward:
annotate MAG215 and liftOver to the reference.</li>
<li>There is missing the annotation of VCF for gene context (BEDtools)
and the prediction of the effect (SnpEff).</li>
</ul></li>
<li><p>09/06/2025</p>
<ul>
<li><strong>Extra plasmid in H01</strong> (contig_4) apparently is
actually a plasmid, relaxes proteins were found and it presents an
<strong>interesting hash similarity to an accessory plasmid</strong> in
other strain of <em>S. meliloti</em> (CP021832). A plasmid matching
using <strong>PLSDB</strong> also showed interesting hash matching with
other accessory plasmids reported in <em>S. meliloti</em> and <em>S.
medicae</em></li>
<li>I annotated with <strong>bakta</strong>, the idea is to see SNPs in
CDS. The extra plasmid has a lot of hypothetical proteins, however it
seems several of them are annotated as such in the RefSeq.</li>
<li>Genome comparison between the GCA genome available at Cerebro with
the H01 (MAG215) presents highly divergence with RM1021. <strong>Are
these true or are miss-assemblies??</strong></li>
</ul></li>
<li><p>05/06/2025</p>
<ul>
<li>I selected Flye assembly genomes. I am trying to find a way to
annotate variants. For example, in a CDS context. What variants are
(non)synonymus? in which genes are? What mutations are point
mutations/truncated protein?</li>
<li>I found an extra plasmid. I will proceed with its identification, I
will use MOB SUITES pipeline.</li>
</ul></li>
</ul>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
