<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>Hunting-a-plasmid</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" />
</head>
<body>

    
    <div class="navbar navbar-static-top">
    <div class="navbar-inner">
      <div class="container">
        <span class="doc-title">Hunting-a-plasmid</span>
        <ul class="nav pull-right doc-info">
                            </ul>
      </div>
    </div>
  </div>
    <div class="container">
    <div class="row">
            <div id="TOC" class="span3">
        <div class="well toc">

        <ul>
        <li><a href="#objective-1-is-it-real"
        id="toc-objective-1-is-it-real">Objective 1: Is it Real?</a>
        <ul>
        <li><a href="#where-was-it-found"
        id="toc-where-was-it-found">Where was it found?</a></li>
        <li><a href="#plasmid-assembly-features"
        id="toc-plasmid-assembly-features">Plasmid assembly
        features</a></li>
        </ul></li>
        <li><a href="#objective-2-is-it-known"
        id="toc-objective-2-is-it-known">Objective 2: Is it known?</a>
        <ul>
        <li><a href="#pmag215-a-novel-plasmid-in-s.-meliloti"
        id="toc-pmag215-a-novel-plasmid-in-s.-meliloti">pMAG215 a novel
        plasmid in <em>S. meliloti</em></a></li>
        <li><a href="#pmag215-annotation"
        id="toc-pmag215-annotation">pMAG215 annotation</a></li>
        </ul></li>
        <li><a href="#objective-3-homology-in-pmag215"
        id="toc-objective-3-homology-in-pmag215">Objective 3: Homology
        in pMAG215</a>
        <ul>
        <li><a href="#kh35c" id="toc-kh35c">KH35c</a></li>
        </ul></li>
        </ul>

        </div>
      </div>
            <div class="span9">

      
      <h1 id="objective-1-is-it-real">Objective 1: Is it Real?</h1>
<p>An extra putative plasmid was identified. The task is to figure out
if it is a plasmid or a sequencing artifact or error.</p>
<h2 id="where-was-it-found">Where was it found?</h2>
<hr />
<p>The plasmid was identified in sample H01 (MAG215). During whole
genome assembly a short sequence was assembled by Hifiasm and Flye.
Their length differ significantly, and only Flye was abled to identify
this sequence as circular (table and pictures below).</p>
<p>Flye was thought for small to medium genomes and has been
characterized for been very good at circularizing bacterial genomes and
plasmids, while Hifiasm was thought for generate haplotype-resolved
assembly of big genomes (created for human genome). Hifiasm has
in-heritage the string graph strategy its author has been developed
since Miniasm and which tended to produce longer/duplicated sequences.
Therefore, in this case, given the similarity between both plasmids
assembled, I prefer continue with Flye’s plasmid assembly.</p>
<table>
<thead>
<tr>
<th style="text-align: center;">Assembler</th>
<th style="text-align: center;">len (bp)</th>
<th>circular</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Flye</td>
<td style="text-align: center;">192,654</td>
<td>Yes</td>
</tr>
<tr>
<td style="text-align: center;">Hifiasm</td>
<td style="text-align: center;">240,319</td>
<td>No</td>
</tr>
</tbody>
</table>
<h2 id="plasmid-assembly-features">Plasmid assembly features</h2>
<hr />
<p>Now, I described the observations that supports the presence of this
plasmid. Let’s start with coverage.</p>
<h3 id="coverage">Coverage</h3>
<p>The extra plasmid presented a coverage congruent with the rest of the
genome. In the next table we can see all the sequences assembled by
Flye, just by looking at the length we can tell which are the main
chromosome, the 2 classic plasmids and the extra plasmid (contig_4).</p>
<table>
<thead>
<tr>
<th>ID</th>
<th>Len</th>
<th>Cov</th>
</tr>
</thead>
<tbody>
<tr>
<td>contig_1</td>
<td>1,689,003</td>
<td>551</td>
</tr>
<tr>
<td>contig_2</td>
<td>3,654,821</td>
<td>558</td>
</tr>
<tr>
<td>contig_3</td>
<td>1,328,226</td>
<td>630</td>
</tr>
<tr>
<td>contig_4</td>
<td>192,654</td>
<td>772</td>
</tr>
</tbody>
</table>
<p>Plasmids can be identified looking for: 1. Relaxases proteins. 2.
Mating Pair Formation system (MPF). 3. OriT. 4. Replication Initiation
Protein (RIP). 5. OriV.</p>
<p><strong><em>MOB Suites</em></strong> was used for typing the putative
plasmid for numeral <strong>1, 2 and 3</strong>. It was abled to found
relaxase proteins. Additionally, they were from a different type
regarding to the other two big plasmids usually found in <em>S.
meliloti</em>. The tool could not find MPF and OriT, reason why it is
expected to be a mobilizable plasmid, but not conjugate (table below).
Additionally, it returned the accession of the plasmid on their database
with which the putative plasmid shares higher number of k-mers and it
turns out to be an accessory plasmid reported in <em>S. meliloti</em>
strain HM006.</p>
<p>Finally, as expected, it was not found any relaxase, OriT or MPF in
the main chromosome.</p>
<table>
<colgroup>
<col style="width: 7%" />
<col style="width: 6%" />
<col style="width: 12%" />
<col style="width: 25%" />
<col style="width: 20%" />
<col style="width: 27%" />
</colgroup>
<thead>
<tr>
<th>ID</th>
<th>len</th>
<th>relaxase_type</th>
<th>relaxase_type_accession(s)</th>
<th>mash_nearest_neighbor</th>
<th>mash_neighbor_identification</th>
</tr>
</thead>
<tbody>
<tr>
<td>Contig_4</td>
<td>192,654</td>
<td>MOBP</td>
<td>CP026529_00118</td>
<td>CP021832</td>
<td>Sinorhizobium meliloti</td>
</tr>
</tbody>
</table>
<p><strong>See complete result here:</strong> <a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/H01/new_plasmid/H01_mobtyper_results.tsv">H01_mobtyper_results.tsv</a></p>
<p><strong><em>OriV-Finder</em></strong> was used to annotate the
putative plasmid for numeral <strong>4 and 5</strong>. The tool was able
to predict an OriV based on the finding of RIPs of type RepA and RepC
(figure 1 and figure 2).</p>
<p><img src="https://webfs/n/projects/jp2992/stowers-notes/photos/oriV.png" class="wikilink"
alt="oriV.png" />
<img src="https://webfs/n/projects/jp2992/stowers-notes/photos/oriV_iterons.png" class="wikilink"
alt="oriV_iterons.png.png"/> <strong>Figure 1.</strong> OriV prediction
close to a GC skew and a RIP (top). Prediction is supported by AT rich
region and prediction of putative Iteron (bottom). | |
<img src="https://webfs/n/projects/jp2992/stowers-notes/photos/repA_new_plasmid.png" class="wikilink"
alt="repA_new_plasmid.png"/>
<img src="https://webfs/n/projects/jp2992/stowers-notes/photos/repC_prediction_new_plasmid.png" class="wikilink"
alt="repC_prediction_new_plasmid.png"/> <strong>Figure 2.</strong> RIPs
found in putative plasmid (contig_4). |</p>
<h1 id="objective-2-is-it-known">Objective 2: Is it known?</h1>
<p>During <a
href="https://webfs/n/projects/jp2992/stowers-notes/Hunting-a-plasmid.html#objective-1-is-it-real">Objective
1</a> the existence of the plasmid was confirmed. The task is to
determine similarity to other accessory plasmids reported in <em>S.
meliloti</em>, in order to find homology and function clues.</p>
<p>For ease, from now on I will refer to the plasmid found in MAG215 as
<strong>pMAG215</strong>.</p>
<h2 id="pmag215-a-novel-plasmid-in-s.-meliloti">pMAG215 a novel plasmid
in <em>S. meliloti</em></h2>
<hr />
<p>According with Siva and Madi there are a couple of accessory plasmid
reported, and some of the most described are <strong>pSmeSM11a, pRmeGR4b
and pTA2</strong> (I could not find the last one in the database).</p>
<p>Therefore, we need to know if the plasmid was reported before and if
not where we can find homology that gives clues about its functions.</p>
<p>Around 753 sequences related with the keywords <em>Sinorhizobium
meliloti[All Fields] AND plasmid[filter]</em> were downloaded from the
NCBI. Sequences were filtered for greater than <strong>10kb</strong> and
lower than <strong>500kb</strong> to remove the two big plasmids, the
chromosome and genes or cassettes. The final dataset contained
<strong>114</strong> sequences. <strong>No similar was found among 75
sequences mapped</strong>, only short regions of probably homology
(Figure 3). <strong>This might indicate is a new plasmid</strong>. The
sequences compiled contained also the plasmids the indicated by Siva:
pSmeSM11a, pRmeGR4b.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/extra_plasmid_H01/aln_homologous/plasmids_smeliloti_filtered.fasta">Dataset
with 114 sequences</a></p>
<p>However, the plasmid showed some degree of homology to other plasmids
reported in Sinorhizobium that might help figuring out functions
associated in other studies. In the pMAG215 annotation section we will
see an example of stretches related to other plasmids in Sinorhizobium,
in that case specifically to <em>S. medicae</em>.</p>
<p><img src="https://webfs/n/projects/jp2992/stowers-notes/photos/plasmids_vs_new_plasmid_asm5.png" class="wikilink"
alt="plasmids_vs_new_plasmid_asm5.png"/> <strong>Figure 3.</strong>
Alignment of 75 plasmids reported for S. meliloti in the NCBI to plasmid
found in MAG215 (contig_4).</p>
<p>Mapping the new plasmid to PLSDB, a plasmid database by the NCBI,
confirmed previous results (Figure 4).
<img src="https://webfs/n/projects/jp2992/stowers-notes/photos/PLSDB.png" class="wikilink"
alt="PLSDB.png"/> <strong>Figure 4.</strong> Results returned by PLSDB
using new plasmid (contig_4) as query. This database contains more
plasmids than the ones we used since it includes different species.</p>
<h2 id="pmag215-annotation">pMAG215 annotation</h2>
<hr />
<h3 id="bakta">BAKTA</h3>
<p>Annotation of the extra plasmid with BAKTA <strong>shows a 34% of
hypothetical proteins</strong>. Figure 5 shows genome map with predicted
CDS and ncRNA and table below shows descriptive stats of annotation.</p>
<p><img src="https://webfs/n/projects/jp2992/stowers-notes/photos/plot-feat.png" class="wikilink"
alt="plot-feat.png"/> <strong>Figure 5</strong>. Genome mapping
summarizes CDS prediction. First ring forward. Second ring</p>
<p><strong>pMAG215 annotation statistics</strong></p>
<table style="width:100%;">
<colgroup>
<col style="width: 8%" />
<col style="width: 5%" />
<col style="width: 16%" />
<col style="width: 6%" />
<col style="width: 16%" />
<col style="width: 4%" />
<col style="width: 22%" />
<col style="width: 19%" />
</colgroup>
<thead>
<tr>
<th>Genome</th>
<th>GC</th>
<th>Coding ratio</th>
<th>ncRNA</th>
<th>ncRNA region</th>
<th>CDS</th>
<th>CDS hypothetical</th>
<th>CDS pseudogene</th>
</tr>
</thead>
<tbody>
<tr>
<td>H01</td>
<td>58.9</td>
<td>78.5</td>
<td>1</td>
<td>1</td>
<td>241</td>
<td>82</td>
<td>5</td>
</tr>
</tbody>
</table>
<h3 id="pgap">PGAP</h3>
<p>The annotation was updated using the NCBI pipeline PGAP. It showed a
significant reduction in annotation. <strong>From now on, we will keep
using this annotation</strong>. This will work as a comparison for
future.</p>
<p><strong>pMAG215 annotation statistics</strong></p>
<table>
<colgroup>
<col style="width: 10%" />
<col style="width: 7%" />
<col style="width: 21%" />
<col style="width: 5%" />
<col style="width: 29%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr>
<th>Genome</th>
<th>GC</th>
<th>Coding ratio</th>
<th>CDS</th>
<th>CDS hypothetical</th>
<th>CDS pseudogene</th>
</tr>
</thead>
<tbody>
<tr>
<td>H01</td>
<td>58.9</td>
<td>77</td>
<td>217</td>
<td>54</td>
<td>42</td>
</tr>
</tbody>
</table>
<p>Using the pMAG215 annotation, once again, we confirm the nature of
this contig as a plasmid. Figure 6 is showing the replication module
<strong><em>RepABC</em></strong> reported in other plasmids of <em>S.
meliloti</em> such as <strong><em>pSmeSM11b</em></strong>, additionally
we see homology in this region with the plasmids aligned previously,
supporting the annotation of these proteins. This was done with BAKTA,
but same results were found in PAGP annotation</p>
<p><img src="https://webfs/n/projects/jp2992/stowers-notes/photos/RIPs_all_plasmids_vs_pmag215.png" class="wikilink"
alt="RIPs_all_plasmids_vs_pmag215.png"/> <strong>Figure 6.</strong>
Region in pMAG215 predicted for RepABC showing homology to other
plasmids in Sinorhizobium.</p>
<h3 id="gene-ontology-clustering">Gene Ontology clustering</h3>
<hr />
<p>This is an example. KEGG was abled to annotate only 21%. 51 out of
241.</p>
<p><img src="https://webfs/n/projects/jp2992/stowers-notes/photos/keeg_annot.png" class="wikilink"
alt="keeg_annot.png"/></p>
<h1 id="objective-3-homology-in-pmag215">Objective 3: Homology in
pMAG215</h1>
<p>During <a
href="https://webfs/n/projects/jp2992/stowers-notes/Hunting-a-plasmid.html#objective-2-is-it-known">Objective
2</a> We aligned pMAG215 with 114 accessory plasmids downloaded from
NCBI and found that pMAG215 had not been reported before as no similar
plasmid existed. However, we found that the pMAG215 overlapped in some
regions with 75 plasmids suggesting regions of homology.</p>
<h2 id="kh35c">KH35c</h2>
<hr />
<p>Siva found some similarities between genes of interest in pMAG215 and
the accessory plasmids of KH35c (pKH35cA). Strain KH35c is important as
Siva has observed <span style="color:#079e15;font-style: italic">some
similar plant phenotype between KH35c and MAG215</span>.</p>
<p>During the mapping to the 114 accessory plasmids of <em>S.
meliloti</em>, pKH35cA was included and actually was among the best
matches. <strong>The next table presents the 10 best
matches</strong>.</p>
<ul>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/extra_plasmid_H01/aln_homologous/plasmids_vs_new_plasmid_blast.tsv">plasmids_vs_new_plasmid_blast.tsv</a>
Presents details about all the accessory plasmids in <em>S.
meliloti</em> with significant matches to pMAG215 using BLAST.</p></li>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/extra_plasmid_H01/aln_homologous/plasmids_vs_new_plasmid_blast.out">plasmids_vs_new_plasmid_blast.out</a>
Presents all the plasmids with significant matches to pMAG215 and their
pairwise alignments.</p></li>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/extra_plasmid_H01/aln_homologous/igv_session.xml">igv_session.xml</a>
This is an IGV session where you can visualize all the significant
matches to pMAG215 with gene annotation of pMAG215. I presents to type
of alignment: one up to 20% of divergence with Minimap2, and second with
local alignment with BLAST.</p></li>
</ul>
<table>
<colgroup>
<col style="width: 10%" />
<col style="width: 77%" />
<col style="width: 5%" />
<col style="width: 7%" />
</colgroup>
<thead>
<tr>
<th>Accession</th>
<th>Description</th>
<th>Score</th>
<th>E-value</th>
</tr>
</thead>
<tbody>
<tr>
<td>CP140898.1</td>
<td>Sinorhizobium medicae strain ml49 plasmid pMl1, complete
sequence</td>
<td>21529</td>
<td>0.0</td>
</tr>
<tr>
<td>CP140966.1</td>
<td>Sinorhizobium medicae strain ml20 plasmid pMl1, complete
sequence</td>
<td>21461</td>
<td>0.0</td>
</tr>
<tr>
<td>CP021832.1</td>
<td>Sinorhizobium meliloti strain HM006 plasmid accessoryA, complete
sequence</td>
<td>17385</td>
<td>0.0</td>
</tr>
<tr>
<td>CP003934.2</td>
<td>Sinorhizobium meliloti GR4 plasmid pRmeGR4a, complete sequence</td>
<td>13372</td>
<td>0.0</td>
</tr>
<tr>
<td>CP021826.1</td>
<td>Sinorhizobium meliloti strain KH35c plasmid accessoryA, complete
sequence</td>
<td>13372</td>
<td>0.0</td>
</tr>
<tr>
<td>CP178489.1</td>
<td>Sinorhizobium meliloti strain RPZ17 plasmid pSkuRPZ17-1, complete
sequence</td>
<td>13104</td>
<td>0.0</td>
</tr>
<tr>
<td>JQ665880.1</td>
<td>Sinorhizobium meliloti plasmid pHRC017, complete sequence</td>
<td>12050</td>
<td>0.0</td>
</tr>
<tr>
<td>CP088116.1</td>
<td>Sinorhizobium meliloti strain RRI128 plasmid pRRI128_3, complete
sequence</td>
<td>11867</td>
<td>0.0</td>
</tr>
<tr>
<td>CP140912.1</td>
<td>Sinorhizobium medicae strain ml29 plasmid pMl5, complete
sequence</td>
<td>10750</td>
<td>0.0</td>
</tr>
<tr>
<td>CP021796.1</td>
<td>Sinorhizobium meliloti strain USDA1157 plasmid accessoryA, complete
sequence</td>
<td>10744</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<h3 id="protein-to-protein-alignment">Protein to protein alignment</h3>
<p><strong><em>Roary</em></strong> with <strong>80% of identity</strong>
was used to clustered similar proteins between pKH35cA and pMAG215. The
next table presents gene shared between them.</p>
<ul>
<li><a
href="https://webfs/n/projects/jp2992/MOLNG4331/extra_plasmid_H01/aln_homologous/kh35c/roary_result_80_pct_1751036011/gene_presence_absence.xlsx">gene_presence_absence.xlsx</a>
Contains all the genes clustered between pKH35cA and pMAG215.</li>
</ul>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
