<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>Min. coverage to get S. meliloti plasmids</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" />
</head>
<body>

    
    <div class="navbar navbar-static-top">
    <div class="navbar-inner">
      <div class="container">
        <span class="doc-title">Min. coverage to get S. meliloti
plasmids</span>
        <ul class="nav pull-right doc-info">
                            </ul>
      </div>
    </div>
  </div>
    <div class="container">
    <div class="row">
            <div id="TOC" class="span3">
        <div class="well toc">

        <ul>
        <li><a href="#antecedents"
        id="toc-antecedents">Antecedents</a></li>
        <li><a href="#in-house-test" id="toc-in-house-test">In-house
        test</a></li>
        </ul>

        </div>
      </div>
            <div class="span9">

      
      <h1 id="antecedents">Antecedents</h1>
<p>The idea is to know how low we can push coverage while getting
complete circular plasmids.</p>
<p>Letâ€™s remember this benchmarking about plasmid recovery by Ryan
Wick.</p>
<p><img 
src=https://webfs/n/projects/jp2992/stowers-notes/photos/plasmid_recovery_RWick.png /></p>
<p><strong>Notice</strong> for our favorite assembly, Flye, the limit is
around 10x.</p>
<p>We can see this event in the collaborator data:
<img height=400 src="https://webfs/n/projects/jp2992/MOLNG4331/collaborator_Smeliloti_hifi/flye_asm/stats_putative_pls/chromosomeCov_vs_accePlasmidCov.png"/>
<strong>Legend</strong>: Number of chromosome assembled circular
(mega-plasmids included).</p>
<p><strong>x-axis</strong>: Mean Chromosome coverage (mega-plasmids
included).</p>
<p><strong>y-axis</strong>: Mean Coverage circular accessory
plasmid.</p>
<p>The plot before show us that considering a median read length between
6-7kb and an uneven coverage between chromosome and plasmid we can get
as low as 10x. Consider here that the minimum plasmid length recovered
was 71,855 bp</p>
<h1 id="in-house-test">In-house test</h1>
<p>Reads for all in-house strains were subsampled to 100x, 50x, 25x</p>
<details>
<summary>
Show code
</summary>
<pre><code>fastq_files=(/n/analysis/Sankari/mm2883/MOLNG-4331/EA156900/LongPlex/longplex_output/MOLNG4331/merged_fastq/*.gz)

task_id=$SLURM_ARRAY_TASK_ID

file=${fastq_files[$task_id]}

fastqc_name=$(basename $file | cut -d &quot;.&quot; -f 1,2 )

genome_size=7000000

median_read_len=$(grep -A 1000 &#39;&gt;&gt;Sequence Length Distribution&#39; /n/analysis/Sankari/mm2883/MOLNG-4331/EA156900/LongPlex/longplex_output/MOLNG4331/merged_fastq/QC/$fastqc_name&quot;_fastqc/fastqc_data.txt&quot; | sed &#39;/^&gt;&gt;END_MODULE/q&#39; | tail -n +3 | head -n -1 | sort -r -n -k2 | head -n1 | cut -f1 | cut -d &quot;-&quot; -f1)

target_coverages=(100 50 25)

for coverage in ${target_coverages[@]}

    do
    
    output_file=$(basename $file | cut -d &quot;.&quot; -f 1,2)
    
    echo &quot;#######&quot;
    
    echo &quot;Considering a median read length of:&quot; $median_read_len
    
    echo &quot;#######&quot;
    
    num_reads=$(( $coverage * $genome_size / $median_read_len ))
    
    echo &quot;For a coverage of $coverage&#39;x&#39; subsampling to this number of reads $num_reads&quot;
    
    conda activate seqtk
    
    seqtk sample $file $num_reads &gt; $(dirname $file)&quot;/&quot;$output_file&quot;_&quot;$coverage&quot;.fastq&quot;

done</code></pre>
</details>
<p>As expected, considering a median read length of 10-11kb, all the
plasmids could be recover as low as 25x.</p>
<p>A02 is the only sample that could not achieve the expected assembly
of 1 chromosome and 2 mega-plasmids, this is due to a repetition in the
chromosome that make Flye to fragment it, however we have seen this
being solve in other assemblers like Hifiasm.</p>
<p>A01, G01 and H01 were the only samples with more than 3 circular
contigs which are the one that carry accessory plasmids, being A01 the
one that carry 2 of them.</p>
<p><img height= 400 src="https://webfs/n/projects/jp2992/stowers-notes/photos/subsampling_test_inhouse.png"/></p>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
