<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>Plasmidome</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" />
</head>
<body>

    
    <div class="navbar navbar-static-top">
    <div class="navbar-inner">
      <div class="container">
        <span class="doc-title">Plasmidome</span>
        <ul class="nav pull-right doc-info">
                            </ul>
      </div>
    </div>
  </div>
    <div class="container">
    <div class="row">
            <div id="TOC" class="span3">
        <div class="well toc">

        <ul>
        <li><a href="#objective" id="toc-objective">Objective</a></li>
        <li><a href="#literature"
        id="toc-literature">Literature</a></li>
        <li><a href="#pre-processing"
        id="toc-pre-processing">Pre-processing</a></li>
        <li><a href="#first-insights" id="toc-first-insights">First
        insights</a></li>
        </ul>

        </div>
      </div>
            <div class="span9">

      
      <h1 id="objective">Objective</h1>
<p>PsymA for years has been consider a biggest hotspot for intra-strain
variability and accessory gene acquisition by horizontal gene transfer,
however it has been found that S. meliloti can also carry accessory
plasmids, plasmids that have been proven to also carry competence genes
affecting symbiosis.</p>
<p>During this project we will collect all the accessory plasmids
available for S. meliloti and their relatedness with other found in the
Rhizobiome.</p>
<h1 id="literature">Literature</h1>
<p>The first step is to get literature, specially metagenomic studies,
describing the bacterial diversity in the Rhizobiome of Medicago sp.
(Alfafa). So we have a starting dataset.</p>
<p><em>Phage-plasmids promote recombination and emergence of phages and
plasmids:</em> here is present an interesting methodology that can work
for us. To evaluate the gene flow between P-P, phages and plasmids they
recognized a classical phylogeny cannot be applied due to their high
level of recombination and fast changing.</p>
<p><em>Comparative analysis of integrative and conjugative mobile
genetic elements in the genus Mesorhizobium</em></p>
<p><em>Plasmid Flux in Escherichia coli ST131 Sublineages, Analyzed by
Plasmid Constellation Network (PLACNET), a New Method for Plasmid
Reconstruction from Whole Genome Sequences</em>: These methodology can
be interesting for us. The limitation here is that <em>E. coli</em> does
not present mega-plasmids, so we cannot use WGS as probably the graph
will be a mess. Can it just used assembled genomes?</p>
<h1 id="pre-processing">Pre-processing</h1>
<p>The PLSDB 2025v and its metadata was downloaded and filtered for only
sequences below 500Kb.</p>
<details>
<summary>
Show command
</summary>
<pre><code>path=/n/projects/jp2992/MOLNG4331/biocomp_tools/plsdb
seqkit seq -M 500000 $path/plsdb_2025.fasta -o $path/plsdb_less_500kb_2025.fasta
</code></pre>
</details>
<h1 id="first-insights">First insights</h1>
<p>Let’s check how many plasmids are available for each specie of
<strong>Sinorhizobium</strong>.</p>
<table>
<colgroup>
<col style="width: 48%" />
<col style="width: 51%" />
</colgroup>
<thead>
<tr>
<th><img src="https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/plasmidome/stats_plsdb/pls_num_sinorhizobium.png"></th>
<th><img src="https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/plasmidome/stats_plsdb/pls_num_rhizobiaceae_genus.png"></th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/plasmidome/stats_plsdb/pls_num_sinorhizobium.png"></p>
<p>Let’s check how many plasmids are available for each genus inside
family <strong>Rhizobiaceae</strong></p>
<p><img src="https://webfs/n/projects/jp2992/MOLNG4331/plasmids_smeliloti/plasmidome/stats_plsdb/pls_num_rhizobiaceae_genus.png"></p>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
