<!doctype html>
<html >
<head>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <!--[if lt IE 9]>
                <script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
        <![endif]-->
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="Content-Style-Type" content="text/css" />

    <!-- <link rel="stylesheet" type="text/css" href="template.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/template.css" />

    <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />

    <script src="https://code.jquery.com/jquery-2.2.1.min.js"></script>
    <!-- <script type='text/javascript' src='menu/js/jquery.cookie.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.hoverIntent.minified.js'></script> -->
    <!-- <script type='text/javascript' src='menu/js/jquery.dcjqaccordion.2.7.min.js'></script> -->

    <!-- <link href="menu/css/skins/blue.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/graphite.css" rel="stylesheet" type="text/css" /> -->
    <!-- <link href="menu/css/skins/grey.css" rel="stylesheet" type="text/css" /> -->
  
    <!-- <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> -->
        
  
    <!-- <script src="script.js"></script> -->
  
    <!-- <script src="jquery.sticky-kit.js "></script> -->
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.cookie.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.hoverIntent.minified.js'></script>
    <script type='text/javascript' src='https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/js/jquery.dcjqaccordion.2.7.min.js'></script>

    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/blue.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/graphite.css" rel="stylesheet" type="text/css" />
    <link href="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/menu/css/skins/grey.css" rel="stylesheet" type="text/css" />
    <link href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" rel="stylesheet" type="text/css" />
  
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/script.js"></script>
  
    <script src="https://cdn.jsdelivr.net/gh/diversen/pandoc-bootstrap-adaptive-template@959c3622/jquery.sticky-kit.js"></script>
    <meta name="generator" content="pandoc" />
  <title>Pan-genome analysis</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="https://webfs/n/projects/jp2992/stowers-notes/easy-pandoc-templates/css/elegant_bootstrap.css" />
</head>
<body>

    
    <div class="navbar navbar-static-top">
    <div class="navbar-inner">
      <div class="container">
        <span class="doc-title">Pan-genome analysis</span>
        <ul class="nav pull-right doc-info">
                            </ul>
      </div>
    </div>
  </div>
    <div class="container">
    <div class="row">
            <div id="TOC" class="span3">
        <div class="well toc">

        <ul>
        <li><a href="#pan-genome-analysis"
        id="toc-pan-genome-analysis">Pan-Genome Analysis</a></li>
        <li><a href="#rm1021-vs-mag215" id="toc-rm1021-vs-mag215">Rm1021
        vs MAG215</a></li>
        <li><a href="#all-vs-all-in-house-strains"
        id="toc-all-vs-all-in-house-strains">All vs All in-house
        strains</a></li>
        <li><a href="#public-mag-strains"
        id="toc-public-mag-strains">Public MAG strains</a>
        <ul>
        <li><a href="#mag-assemblies-metadata"
        id="toc-mag-assemblies-metadata">MAG assemblies
        metadata</a></li>
        <li><a href="#susceptible-mag-strains"
        id="toc-susceptible-mag-strains">Susceptible MAG
        strains</a></li>
        <li><a href="#mag215-vs-susceptible-mag-pan-genome"
        id="toc-mag215-vs-susceptible-mag-pan-genome">MAG215 vs
        Susceptible MAG pan-genome</a></li>
        <li><a href="#pmag215-containment-in-other-mag-strains"
        id="toc-pmag215-containment-in-other-mag-strains">pMAG215
        containment in other MAG strains</a></li>
        </ul></li>
        </ul>

        </div>
      </div>
            <div class="span9">

      
      <hr />
<h1 id="pan-genome-analysis">Pan-Genome Analysis</h1>
<hr />
<blockquote>
<p>[!NOTE]</p>
<p>Everything related with pan-genome will be stored in the following
folder: <a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/">pan-genome</a></p>
</blockquote>
<h1 id="rm1021-vs-mag215">Rm1021 vs MAG215</h1>
<hr />
<p>With the observed conservation variation between MAG215 and Sankari’s
RM1021 during genome synteny analysis, we decided to run a pan-genome
analysis with <strong><em>Roary</em></strong>.</p>
<p>A pan-genome analysis can give us genes comparison between the
reference and the MAG215 strain. We found the core genome for these two
strains is made of <strong>74.2%</strong> leaving a
<strong>25.8%</strong> of variable genes (Shell genes).</p>
<table>
<thead>
<tr>
<th>Pan-genome</th>
<th># genes</th>
</tr>
</thead>
<tbody>
<tr>
<td>Core genes (99% &lt;= strains &lt;= 100%)</td>
<td>5447</td>
</tr>
<tr>
<td>Shell genes (15% &lt;= strains &lt; 95%)</td>
<td>1888</td>
</tr>
<tr>
<td>Total genes (0% &lt;= strains &lt;= 100%)</td>
<td>7335</td>
</tr>
</tbody>
</table>
<p><strong>After discussion with Siva, we decided to try lower
thresholds of protein identity</strong> in order to try to reduce the
number of unique genes assigned to each strain (the 1888 genes above).
However after trying 90, 80 and 50% few changes in number of unique
genes were observed.</p>
<h4 id="important-files">Important files</h4>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/gene_presence_absence_all_pct.xlsx">gene_presence_absence_all_pct.xlsx</a>
contains the clustering for both strains MAG215 and Rm1021 using 95, 90,
80 and 50% identity percentages.</p>
<p><strong><em>Clustering at 50%</em></strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/roary_result_50_pct/gene_presence_absence.csv">gene_presence_absence.csv</a>
is the original file which contains a table with all the genes clustered
for both genomes.</p>
<p><strong><em>Clustering at 80%</em></strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/roary_result_80_pct/gene_presence_absence.csv">gene_presence_absence.csv</a>
is the original file which contains a table with all the genes clustered
for both genomes.</p>
<p><strong><em>Clustering at 90%</em></strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/roary_result_90_pct/gene_presence_absence.csv">gene_presence_absence.csv</a>
is the original file which contains a table with all the genes clustered
for both genomes.</p>
<p><strong><em>Clustering at 95%</em></strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/roary_result_95_pct/gene_presence_absence_all_genome.csv">gene_presence_absence_all_genome.csv</a>
is the original file which contains a table with all the genes clustered
for both genomes.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/roary_result_95_pct/gene_presence_absence.xlsx">gene_presence_absence.xlsx</a>
is a modified form which contains two extra spreadsheets for accessory
genes for MAG215 and RM1021 with an additional column (contig loc)
indicating where is the gene coming from.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/roary_result/pan_genome_reference.fa">pan_genome_reference.fa</a>
contains a single representative nucleotide sequence from each of the
clusters in the pan genome (core and accessory). The name of each
sequence is the source sequence ID followed by the cluster it came
from.</p>
<p><strong>Some counts for the accessory genes:</strong></p>
<table>
<thead>
<tr>
<th>Strain</th>
<th>Accessory genes</th>
<th>Hypothetical proteins</th>
<th>Hits with no ref</th>
</tr>
</thead>
<tbody>
<tr>
<td>RM1021</td>
<td>849</td>
<td>141</td>
<td>4</td>
</tr>
<tr>
<td>MAG215</td>
<td>1039</td>
<td>255</td>
<td>7</td>
</tr>
<tr>
<td>Total</td>
<td>1888</td>
<td>396</td>
<td>11</td>
</tr>
</tbody>
</table>
<h1 id="all-vs-all-in-house-strains">All vs All in-house strains</h1>
<hr />
<p>Using the strains of <em>S .meliloti</em> assembled previously (see
this <a
href="https://webfs/n/projects/jp2992/stowers-notes/Project-report-cbio.mm2883.102.html#descriptive-stats">table</a>),
we performed a pan-genome analysis using
<strong><em>Roary</em></strong>. Results are showed below for 50% and
80%.</p>
<p><strong>50%</strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/between_in_house_strains//roary_result_50_pct_1751912799/pangenome_matrix.png">pangenome_matrix.png</a>
shows a matrix of presence/absence of accessory genome next to the
dendrogram grouping samples by accessory genome.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/between_in_house_strains/roary_result_50_pct_1751912799/set_difference_unique_set_one_statistics.csv">set_difference_unique_set_one_statistics.csv</a>
presents genes unique to MAG215 (H01).</p>
<table>
<thead>
<tr>
<th>Gene Category</th>
<th>Strain Frequency Range</th>
<th>Number of Genes</th>
</tr>
</thead>
<tbody>
<tr>
<td>Core genes</td>
<td>99% ≤ strains ≤ 100%</td>
<td>4933</td>
</tr>
<tr>
<td>Soft core genes</td>
<td>95% ≤ strains &lt; 99%</td>
<td>0</td>
</tr>
<tr>
<td>Shell genes</td>
<td>15% ≤ strains &lt; 95%</td>
<td>2348</td>
</tr>
<tr>
<td>Cloud genes</td>
<td>0% ≤ strains &lt; 15%</td>
<td>1981</td>
</tr>
<tr>
<td>Total genes</td>
<td>0% ≤ strains ≤ 100%</td>
<td>9262</td>
</tr>
</tbody>
</table>
<p><strong>80%</strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/between_in_house_strains/roary_result_80_pct_1751912805/pangenome_matrix.png">pangenome_matrix.png</a>
shows a matrix of presence/absence of accessory genome next to the
dendrogram grouping samples by accessory genome.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/between_in_house_strains/roary_result_80_pct_1751912805/set_difference_unique_set_one_statistics.csv">set_difference_unique_set_one_statistics.csv</a>
presents genes unique to MAG215 (H01).</p>
<table>
<thead>
<tr>
<th>Gene Category</th>
<th>Strain Frequency Range</th>
<th>Number of Genes</th>
</tr>
</thead>
<tbody>
<tr>
<td>Core genes</td>
<td>99% ≤ strains ≤ 100%</td>
<td>4920</td>
</tr>
<tr>
<td>Soft core genes</td>
<td>95% ≤ strains &lt; 99%</td>
<td>0</td>
</tr>
<tr>
<td>Shell genes</td>
<td>15% ≤ strains &lt; 95%</td>
<td>2324</td>
</tr>
<tr>
<td>Cloud genes</td>
<td>0% ≤ strains &lt; 15%</td>
<td>2162</td>
</tr>
<tr>
<td>Total genes</td>
<td>0% ≤ strains ≤ 100%</td>
<td>9406</td>
</tr>
</tbody>
</table>
<p>Once again, like during QUAST assessment, in terms of accessory
genome we see F1 as the genome that shares more features with MAG215
(H01).</p>
<h1 id="public-mag-strains">Public MAG strains</h1>
<hr />
<p>We observed no significant reduction between MAG215 and Rm1021 (the
max. reduction was 200 genes between 95% and 50%), one reason for this
may be the divergence between strains. Therefore, <strong>we will try to
use MAG strains from other authors for which Madi had information about
which ones had susceptibility</strong> in a attempt to make a
reconstruction of MAG215 genome using susceptible strains to find unique
zones in MAG215 involved with resistance.</p>
<h2 id="mag-assemblies-metadata">MAG assemblies metadata</h2>
<p>Bioinformatician Daniel provided me with the MAG genomes downloaded
for a previous work. According to Daniel’s report, Information about
these strains were retrieved from <a
href="https://doi.org/10.1098/rspb.2022.0477">Batstone et. al
2022</a></p>
<p>Below, important documents hosting genomes and metadata:</p>
<ul>
<li><a
href="https://webfs/n/core/Bioinformatics/analysis/Sankari/mm2883/cbio.mm2883.101/data/MAG_key.tsv">Accession
per MAG strain.tsv</a>: Accession code for the MAG strains genomes
included in the folder below (87).</li>
<li><a
href="https://webfs/n/core/Bioinformatics/analysis/Sankari/mm2883/cbio.mm2883.101/data/Genbank_09.24">Smeliloti
genomes</a>: Folder with several <em>S. meliloti</em> genomes retrieved
form the NCBI (471).</li>
<li><a
href="https://webfs/n/core/Bioinformatics/analysis/Sankari/mm2883/cbio.mm2883.101/data/MAG_metadata.tsv">MAG
strains metadata.tsv</a> Contains information about MAG strains genome
assemblies.</li>
<li><a
href="https://webfs/n/core/Bioinformatics/analysis/Sankari/mm2883/cbio.mm2883.101/code/report.html">Daniel’s
report</a>: Report that Daniel produced for a previous work with Siva.
The are some information about the MAG strains.</li>
<li><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/MAG_assembly_metadata.tsv">MAG_assembly_metadata.tsv</a>
Contains extra metadata about the MAG assemblies.</li>
</ul>
<blockquote>
<p>[!NOTE] During processing of <a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/MAG_genomes/MAG_assembly_metadata.tsv">MAG_assembly_metadata.tsv</a>
we realized there were some genomes with completeness bellow 99, in
order to establish a good core-genome we need assemblies as complete as
possible. During this <a
href="https://webfs//n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/MAG_genomes/mag_metadata.ipynb">notebook</a>
I filtered to keep only genomes with completeness &gt; 99% according to
CheckM and saved in <a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/MAG_genomes/keys_best_asm.tsv">keys_best_asm.tsv</a>
. <strong>It was retained 77 out of 89 MAG genomes</strong>. The
retained genomes were stored at <a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/best_asm_mag/">best_asm_mag</a></p>
</blockquote>
<h3 id="best-mag-assemblies-stats">Best MAG assemblies stats</h3>
<p>Below, Histograms showing distribution of completeness (percentage),
number of contigs assembled (count), N50 and L50 metric:</p>
<ul>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/best_asm_mag/hist_completeness.png">hist_completeness.png</a></p></li>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/best_asm_mag/hist_num_contigs.png">hist_num_contigs.png</a></p></li>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/best_asm_mag/hist_contigN50.png">hist_contigN50.png</a></p></li>
<li><p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/best_asm_mag/hist_contigL50.png">hist_contigL50.png</a></p></li>
</ul>
<h2 id="susceptible-mag-strains">Susceptible MAG strains</h2>
<p>Madi has identified the following susceptible MAG strains:</p>
<table>
<colgroup>
<col style="width: 5%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
</colgroup>
<thead>
<tr>
<th></th>
<th>1</th>
<th>2</th>
<th>3</th>
<th>4</th>
<th>5</th>
<th>6</th>
<th>7</th>
<th>8</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>MAG5</td>
<td>MAG10</td>
<td>MAG16</td>
<td>MAG21</td>
<td>MAG24</td>
<td>MAG27</td>
<td>MAG73</td>
<td>MAG88</td>
</tr>
<tr>
<td>2</td>
<td>MAG97</td>
<td>MAG121</td>
<td>MAG123</td>
<td>MAG41</td>
<td>MAG42</td>
<td>MAG48</td>
<td>MAG145</td>
<td>MAG157</td>
</tr>
<tr>
<td>3</td>
<td>MAG169</td>
<td>MAG182</td>
<td>MAG183</td>
<td>MAG191</td>
<td>MAG199</td>
<td>MAG200</td>
<td>MAG201</td>
<td>MAG220</td>
</tr>
<tr>
<td>4</td>
<td>MAG221</td>
<td>MAG225</td>
<td>MAG230</td>
<td>MAG242</td>
<td>MAG245</td>
<td>MAG258</td>
<td>MAG259</td>
<td>MAG278</td>
</tr>
<tr>
<td>5</td>
<td>MAG282</td>
<td>MAG286</td>
<td>MAG313</td>
<td>MAG320</td>
<td>MAG322</td>
<td>MAG325</td>
<td>MAG342</td>
<td>MAG358</td>
</tr>
<tr>
<td>6</td>
<td>MAG373</td>
<td>MAG384</td>
<td>MAG386</td>
<td>MAG406</td>
<td>MAG417</td>
<td>MAG420</td>
<td>MAG513</td>
<td>MAG514</td>
</tr>
<tr>
<td>7</td>
<td>MAG523</td>
<td>MAG530</td>
<td>MAG533</td>
<td>MAG539</td>
<td>MAG540</td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Since we have 86 susceptible and previously we mentioned earlier that
we filtered the dataset of MAG genomes to 77. At the end, <strong>we
were abled to retrieved 52 MAG susceptible genomes with good quality
metrics</strong>. Below some histograms showing the distribution of
different quality metrics across these genomes.</p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/MAG_genomes/susceptible_mag_genomes/descrip_stats_histo.png">descrip_stats_histo.png</a></p>
<h2 id="mag215-vs-susceptible-mag-pan-genome">MAG215 vs Susceptible MAG
pan-genome</h2>
<p>The idea is to check for gene unique to MAG215 that might confer
resistance to the peptide. To do this I used PPanGGoLin, a tool to
create and manipulate prokaryotic pan-genomes.</p>
<p>The H01 unique genes were extracted with a custom python script <a
href="https://webfs/n/projects/jp2992/MOLNG4331/code/parse_ppanggolin_output.ipynb">parse_ppanggolin_output.ipynb</a></p>
<p><strong>To cluster protein families I set 80% coverage and 80%
identity.</strong></p>
<p><a
href="https://webfs/n/projects/jp2992/MOLNG4331/genomes/genome_comparison/pan_genome/susceptible_MAG_vs_mag215/ppanggolin_results_80pct/H01_unique_genes_extended.tsv">H01_unique_genes_extended.tsv</a>
contains the genes found only in H01 including other information
provided by tool, such as RGP and genes neighboring.</p>
<h3 id="genome-similarity-between-mag215-and-mag-strains">Genome
similarity between MAG215 and MAG strains</h3>
<p>Let’s check similarity between MAG215 and the others 88 MAG genomes
using k-mer analysis with <strong><em>SourMash</em></strong>. The
following analysis basically gives us <strong>insights</strong> of how
much of <em>X</em> genome in shared with <em>Y</em> genome. In this
case, we are sampling a k-mer of 31 every 100 k-mers (scaled=100).</p>
<table>
<thead>
<tr>
<th>Similarity</th>
<th>Match</th>
</tr>
</thead>
<tbody>
<tr>
<td>100.0%</td>
<td>H01_asm.fasta</td>
</tr>
<tr>
<td>99.7%</td>
<td>GCA_033715675.1 strain=215</td>
</tr>
<tr>
<td>72.7%</td>
<td>GCA_033716695.1 strain=386</td>
</tr>
<tr>
<td>72.4%</td>
<td>GCA_033714725.1 strain=749C</td>
</tr>
<tr>
<td>72.3%</td>
<td>GCA_033714075.1 strain=342</td>
</tr>
<tr>
<td>71.6%</td>
<td>GCA_033713535.1 strain=146</td>
</tr>
<tr>
<td>71.3%</td>
<td>GCA_033716305.1 strain=748A</td>
</tr>
<tr>
<td>71.3%</td>
<td>GCA_033714695.1 strain=733A</td>
</tr>
<tr>
<td>71.2%</td>
<td>GCA_033712675.1 strain=24</td>
</tr>
<tr>
<td>71.2%</td>
<td>GCA_033715705.1 strain=243A</td>
</tr>
</tbody>
</table>
<h2 id="pmag215-containment-in-other-mag-strains">pMAG215 containment in
other MAG strains</h2>
<p>Using <strong><em>SourMash</em></strong>, we looked for the presence
of the extra plasmid found in MAG215 in the other MAG strains. We see
the maximum containment was from MAG97.</p>
<table>
<thead>
<tr>
<th>Similarity</th>
<th>Match</th>
</tr>
</thead>
<tbody>
<tr>
<td>100.0%</td>
<td>H01_asm.fasta</td>
</tr>
<tr>
<td>98.6%</td>
<td>GCA_033715675.1 strain=215</td>
</tr>
<tr>
<td>34.9%</td>
<td>GCA_033716795.1 strain=97</td>
</tr>
<tr>
<td>33.8%</td>
<td>GCA_033714725.1 strain=749C</td>
</tr>
<tr>
<td>32.7%</td>
<td>GCA_033712485.1 strain=18</td>
</tr>
<tr>
<td>30.0%</td>
<td>GCA_033716245.1 strain=743</td>
</tr>
<tr>
<td>20.0%</td>
<td>GCA_033714785.1 strain=180</td>
</tr>
<tr>
<td>19.4%</td>
<td>GCA_033712715.1 strain=50</td>
</tr>
<tr>
<td>19.3%</td>
<td>GCA_033713635.1 strain=228A</td>
</tr>
<tr>
<td>19.1%</td>
<td>GCA_033712675.1 strain=24</td>
</tr>
</tbody>
</table>
<p>In this <a
href="https://webfs/n/projects/jp2992/MOLNG4331/flye/pan-genome_analysis/MAG_genomes/quast/icarus_viewers/alignment_viewer.html">QUAST</a>
comparison it was aligned strain <strong>MAG386</strong> (previous
table), <strong>B02</strong> (Sankari lab’s rm1021) and
<strong>MAG97</strong>. It allows us to see the amount of plasmid found
in MAG97 and the genome coverage for its closest MAG strain and the
classic Rm1021.</p>
            </div>
    </div>
  </div>
  <script src="https://vjs.zencdn.net/5.4.4/video.js"></script>

</body>
</html>
